packages/fastify-renderer/src/node/renderers/react/StaticWorker.import.js

```diff
@@ -0,0 +1,14 @@
+try {
+  const tsWorkerPath = require.resolve('./StaticWorker.ts')
+
+  // Context: https://github.com/TypeStrong/ts-node/issues/676#issuecomment-470898116
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  require('ts-node').register({
+    // Disable type-checking
+    transpileOnly: true,
+  })
+  require(tsWorkerPath)
+} catch {
```

Comment:
I think this catch is going to catch any errors importing the actual target file, like syntax errors or runtime issues. Could we drive the decision to use ts-node or not with an environment variable or some other explicit instruction?

Reply:
I think magic flags should be avoided imo, especially given that it only matters in development of this package itself.

`.ts` files will never appear in the built package, which will make this always throw when distributed, and always load the js file instead.

It's safe in that respect.

I can do it based on the existence of a `.ts` file instead. Would that be better?

Reply:
@airhorns Actually this can be handled really nicely via vite, but I can't do that right now because the test app doesn't use vite. It could be a small project for later if you like.